pkg/controller/component/kbagent.go

```diff
@@ -278,47 +307,61 @@ func customExecActionImageNContainer(synthesizedComp *SynthesizedComponent) (str
 		actions = append(actions, &synthesizedComp.LifecycleActions.RoleProbe.Action)
 	}
 
-	var image, container string
+	// TODO: Discussion is needed for the definition of action name
+	actionNameMap := map[*appsv1alpha1.Action]string{
+		synthesizedComp.LifecycleActions.PostProvision:     "postpr",
+		synthesizedComp.LifecycleActions.PreTerminate:      "preter",
+		synthesizedComp.LifecycleActions.Switchover:        "switch",
+		synthesizedComp.LifecycleActions.MemberJoin:        "mbrin",
+		synthesizedComp.LifecycleActions.MemberLeave:       "mbrlv",
+		synthesizedComp.LifecycleActions.Readonly:          "readol",
+		synthesizedComp.LifecycleActions.Readwrite:         "readwr",
+		synthesizedComp.LifecycleActions.DataDump:          "datadp",
+		synthesizedComp.LifecycleActions.DataLoad:          "datald",
+		synthesizedComp.LifecycleActions.Reconfigure:       "reconf",
+		synthesizedComp.LifecycleActions.AccountProvision:  "accpr",
+		&synthesizedComp.LifecycleActions.RoleProbe.Action: "rolepb",
+	}
+
+	var images []string
+	var containerNames []string
+	var actionNames []string
 	for _, action := range actions {
 		if action == nil || action.Exec == nil {
 			continue
 		}
-		if action.Exec.Image != "" {
-			if len(image) > 0 && image != action.Exec.Image {
-				return "", nil, fmt.Errorf("only one exec image is allowed in lifecycle actions")
-			}
-			image = action.Exec.Image
+		image := action.Exec.Image
+		containerName := action.Exec.Container
+		if len(image) == 0 && len(containerName) == 0 {
```

Comment:
according to api-doc, if action.Exec.Container is specified, it means will be exectued inside the container, so there is no need to create a new container for it.